libclc/amdgcn-amdhsa/libspirv/atomic/atomic_add.cl

```diff
@@ -28,6 +56,10 @@ AMDGPU_ATOMIC(_Z21__spirv_AtomicFAddEXT, float, f, __hip_atomic_fetch_add)
       _Z21__spirv_AtomicFAddEXTP##AS_MANGLED##dN5__spv5Scope4FlagENS##SUB1##_19MemorySemanticsMask4FlagEd(                      \
           volatile AS double *p, enum Scope scope,                                                                              \
           enum MemorySemanticsMask semantics, double val) {                                                                     \
+    if (CHECK) {                                                                                                                \
+      double ret = NEW_BUILTIN(p, val);                                                                                         \
+      return *(double *)&ret;                                                                                                   \
```

Comment:
Why do we need the cast and the dereference?

Reply:
Oops some copypasta here. Have refactored